pkg/util/testutil/patternscanner.go

```diff
@@ -15,32 +15,49 @@ import (
 	"testing"
 )
 
+// NoPattern is a sugar syntax for empty pattern
+var NoPattern *regexp.Regexp
+
 // PatternScanner is a helper to scan logs for a given pattern.
 type PatternScanner struct {
-	// The log pattern to match on
-	pattern *regexp.Regexp
+	// The log pattern to match on validating successful start
+	startPattern *regexp.Regexp
+	// The log pattern to match on validating successful finish. This is optional
+	finishPattern *regexp.Regexp
```

Comment:
recommendation - instead of supporting only start and finish, support an ordered array of regexes

Reply:
i thought about it, but it felt more complex code and given currently we don't have a use-case for more than 2 (actually the `finishPattern` is added, because we have a need for that for the GPU UTs)

i will follow the "rule of 3" and refactor the code in case another regex will be required